### ISSUE TITLE
Errors trying to run using-wordpress example

### ISSUE THREAD

**Issue by ilyabo** [gatsbyjs/gatsby#2621]
I am trying to run the wordpress API example. Using a fresh clone of master with latest gatsby. It fails with multiple errors. At least some of them are apparently caused by unauthorized API calls. Is it just because the plugin configuration is missing  the auth section?

```
success delete html files from previous builds — 0.011 s
success open and validate gatsby-config.js — 0.006 s
success copy gatsby files — 0.025 s
success onPreBootstrap — 0.009 s
⠁ source and transform nodes -> wordpress__POST fetched : 2
⡀ source and transform nodes -> wordpress__PAGE fetched : 1
⠐ source and transform nodes -> wordpress__wp_media fetched : 2
⠂ source and transform nodes -> wordpress__wp_types fetched : 1
⠐ source and transform nodes -> wordpress__wp_statuses fetched : 1
⠁ source and transform nodes -> wordpress__wp_taxonomies fetched : 1
⢀ source and transform nodes -> wordpress__CATEGORY fetched : 3
⠈ source and transform nodes -> wordpress__TAG fetched : 2
⠄ source and transform nodes -> wordpress__wp_users fetched : 1
⠠ source and transform nodesThe server response was "401 Unauthorized"
Inner exception message : "You are not currently logged in."
⠁ source and transform nodes -> wordpress__wp_comments fetched : 0
⡀ source and transform nodesThe server response was "403 Forbidden"
Inner exception message : "Sorry, you are not allowed to do that."
⠐ source and transform nodesThe server response was "404 Not Found"
Inner exception message : "No route was found matching the URL and request method"
success source and transform nodes — 5.794 s
success building schema — 0.255 s
success createLayouts — 0.020 s
success createPages — 0.025 s
success createPagesStatefully — 0.007 s
success onPreExtractQueries — 0.003 s
GraphQL Error Unknown type "WordPressAcf_post_photo".

  file: /Users/ilyabo/Workspace/_pantry/gatsby/examples/using-wordpress/src/templates/post.js

   1 | 
   2 |   query currentPostQuery($id: String!) {
   3 |     wordpressPost(id: { eq: $id }) {
   4 |       title
   5 |       content
   6 |       ...PostIcons
   7 |       acf {
   8 |         page_builder_post {
   9 |           __typename
> 10 |           ... on WordPressAcf_post_photo {
     |                  ^
  11 |             photo {
  12 |               localFile {
  13 |                 childImageSharp {
  14 |                   sizes(maxWidth: 680) {
  15 |                     ...GatsbyImageSharpSizes
  16 |                   }
  17 |                 }
  18 |               }
  19 |             }
  20 |           }

success extract queries from components — 0.222 s
success run graphql queries — 0.009 s
success write out page data — 0.012 s
success write out redirect data — 0.004 s
success update schema — 0.217 s
success onPostBootstrap — 0.001 s

info bootstrap finished - 8.446 s

error There was an error compiling the html.js component for the development server.

See our docs page on debugging HTML builds for help https://goo.gl/yL9lND


  Error: Module build failed: TypeError: [BABEL] /Users/ilyabo/Workspace/_pantry/gatsby/  examples/using-wordpress/.cache/develop-static-entry.js: Invalid options type for fore  ign
  
  - logger.js:39 Logger.error
    [Workspace]/[babel-core]/lib/transformation/file/logger.js:39:11
  
  - option-manager.js:243 OptionManager.mergeOptions
    [Workspace]/[babel-core]/lib/transformation/file/options/option-manager.js:243:16
  
  - option-manager.js:349 
    [Workspace]/[babel-core]/lib/transformation/file/options/option-manager.js:349:14
  
  - option-manager.js:375 
    [Workspace]/[babel-core]/lib/transformation/file/options/option-manager.js:375:22
  
  - Array.map
  
  - option-manager.js:364 OptionManager.resolvePresets
    [Workspace]/[babel-core]/lib/transformation/file/options/option-manager.js:364:20
  
  - option-manager.js:348 OptionManager.mergePresets
    [Workspace]/[babel-core]/lib/transformation/file/options/option-manager.js:348:10
  
  - option-manager.js:307 OptionManager.mergeOptions
    [Workspace]/[babel-core]/lib/transformation/file/options/option-manager.js:307:14
  
  - option-manager.js:465 OptionManager.init
    [Workspace]/[babel-core]/lib/transformation/file/options/option-manager.js:465:10
  
  - index.js:194 File.initOptions
    [Workspace]/[babel-core]/lib/transformation/file/index.js:194:75
  
  - index.js:123 new File
    [Workspace]/[babel-core]/lib/transformation/file/index.js:123:22
  
  - pipeline.js:45 Pipeline.transform
    [Workspace]/[babel-core]/lib/transformation/pipeline.js:45:16
  
  - index.js:14 transpile
    [Workspace]/[babel-loader]/index.js:14:22
  
  - fs-cache.js:145 
    [Workspace]/[babel-loader]/lib/fs-cache.js:145:18
  
  - fs-cache.js:28 ReadFileContext.callback
    [Workspace]/[babel-loader]/lib/fs-cache.js:28:23
  

```

**Comment by KyleAMathews** [comment]
The auth isn't required so that shouldn't be a problem but maybe? Not home but can debug later. Lemme know if you figure out something.

**Comment by ilyabo** [comment]
I now tried to create a new website on wordpress.com and added a developer app there. I added the auth section to the gatsby wordpress plugin conf accordingly and set `hostingWPCOM: true`. Now more requests get through, but still it fails on photos:

```
success delete html files from previous builds — 0.008 s
success open and validate gatsby-config.js — 0.004 s
info One or more of your plugins have changed since the last time you ran Gatsby. As
a precaution, we're deleting your site's cache to ensure there's not any stale
data
success copy gatsby files — 0.042 s
success onPreBootstrap — 0.008 s
⠁ source and transform nodes -> wordpress__POST fetched : 3
⠠ source and transform nodes -> wordpress__PAGE fetched : 5
⠈ source and transform nodes -> wordpress__wp_media fetched : 9
⠄ source and transform nodes -> wordpress__wp_jp_pay_order fetched : 0
⠈ source and transform nodes -> wordpress__wp_jp_pay_product fetched : 0
⡀ source and transform nodes -> wordpress__wp_feedback fetched : 0
⠐ source and transform nodes -> wordpress__wp_types fetched : 1
⠄ source and transform nodes -> wordpress__wp_statuses fetched : 1
⠈ source and transform nodes -> wordpress__wp_taxonomies fetched : 1
⠄ source and transform nodes -> wordpress__CATEGORY fetched : 1
⠐ source and transform nodes -> wordpress__TAG fetched : 1
⡀ source and transform nodes -> wordpress__wp_users fetched : 1
⠁ source and transform nodes -> wordpress__wp_me fetched : 1
⠈ source and transform nodes -> wordpress__wp_comments fetched : 0
⠐ source and transform nodes -> wordpress__wp_settings fetched : 1
⡀ source and transform nodes -> wordpress__wp_jetpack_testimonial fetched : 2
⠂ source and transform nodesThe server response was "404 Not Found"
Inner exception message : "No route was found matching the URL and request method"
success source and transform nodes — 7.167 s
success building schema — 0.296 s
success createLayouts — 0.042 s
success createPages — 0.040 s
success createPagesStatefully — 0.006 s
success onPreExtractQueries — 0.005 s
GraphQL Error Unknown type "WordPressAcf_post_photo".

  file: /Users/ilyabo/Workspace/fsb/src/templates/post.js

   1 | 
   2 |   query currentPostQuery($id: String!) {
   3 |     wordpressPost(id: { eq: $id }) {
   4 |       title
   5 |       content
   6 |       ...PostIcons
   7 |       acf {
   8 |         page_builder_post {
   9 |           __typename
> 10 |           ... on WordPressAcf_post_photo {
     |                  ^
  11 |             photo {
  12 |               localFile {
  13 |                 childImageSharp {
  14 |                   sizes(maxWidth: 680) {
  15 |                     ...GatsbyImageSharpSizes
  16 |                   }
  17 |                 }
  18 |               }
  19 |             }
  20 |           }

success extract queries from components — 0.217 s
GraphQL Error Unknown type "WordPressAcf_post_photo".

  file: /Users/ilyabo/Workspace/fsb/src/templates/post.js

   1 | 
   2 |   query currentPostQuery($id: String!) {
   3 |     wordpressPost(id: { eq: $id }) {
   4 |       title
   5 |       content
   6 |       ...PostIcons
   7 |       acf {
   8 |         page_builder_post {
   9 |           __typename
> 10 |           ... on WordPressAcf_post_photo {
     |                  ^
  11 |             photo {
  12 |               localFile {
  13 |                 childImageSharp {
  14 |                   sizes(maxWidth: 680) {
  15 |                     ...GatsbyImageSharpSizes
  16 |                   }
  17 |                 }
  18 |               }
  19 |             }
  20 |           }

success run graphql queries — 0.098 s
success write out page data — 0.004 s
success write out redirect data — 0.001 s
success update schema — 0.169 s
success onPostBootstrap — 0.001 s

info bootstrap finished - 11.02 s

error There was an error compiling the html.js component for the development server.

See our docs page on debugging HTML builds for help https://goo.gl/yL9lND


  Error: Module build failed: TypeError: [BABEL] /Users/ilyabo/Workspace/fsb/.cache/develop-static-entry.js: Invalid options type for foreign
  
  - logger.js:39 Logger.error
    [Workspace]/[babel-core]/lib/transformation/file/logger.js:39:11
  
  - option-manager.js:243 OptionManager.mergeOptions
    [Workspace]/[babel-core]/lib/transformation/file/options/option-manager.js:243:16
  
  - option-manager.js:349 
    [Workspace]/[babel-core]/lib/transformation/file/options/option-manager.js:349:14
  
  - option-manager.js:375 
    [Workspace]/[babel-core]/lib/transformation/file/options/option-manager.js:375:22
  
  - Array.map
  
  - option-manager.js:364 OptionManager.resolvePresets
    [Workspace]/[babel-core]/lib/transformation/file/options/option-manager.js:364:20
  
  - option-manager.js:348 OptionManager.mergePresets
    [Workspace]/[babel-core]/lib/transformation/file/options/option-manager.js:348:10
  
  - option-manager.js:307 OptionManager.mergeOptions
    [Workspace]/[babel-core]/lib/transformation/file/options/option-manager.js:307:14
  
  - option-manager.js:465 OptionManager.init
    [Workspace]/[babel-core]/lib/transformation/file/options/option-manager.js:465:10
  
  - index.js:194 File.initOptions
    [Workspace]/[babel-core]/lib/transformation/file/index.js:194:75
  
  - index.js:123 new File
    [Workspace]/[babel-core]/lib/transformation/file/index.js:123:22
  
  - pipeline.js:45 Pipeline.transform
    [Workspace]/[babel-core]/lib/transformation/pipeline.js:45:16
  
  - index.js:14 transpile
    [Workspace]/[babel-loader]/index.js:14:22
  
  - fs-cache.js:145 
    [Workspace]/[babel-loader]/lib/fs-cache.js:145:18
  
  - fs-cache.js:28 ReadFileContext.callback
    [Workspace]/[babel-loader]/lib/fs-cache.js:28:23


```

**Comment by ilyabo** [comment]
This could probably do with the fact that there is no ACF on this newly created wordpress site. I tried disabling setting `useACF: false` but it doesn't solve the error.

**Comment by ilyabo** [comment]
BTW, I am a bit confused about the use of this plugin, maybe you can answer it. If I understand correctly GraphQL which wraps the Wordpress API is used at build time only. Does that mean that if the data on Wordpress changes I would need to rebuild and redeploy the website so that it reflects the changes?

**Comment by KyleAMathews** [comment]
>Does that mean that if the data in Wordpress changes I would need to rebuild and redeploy the website so that it reflects the changes?

Yeah! Gatsby produces a static site. There are ways to automate this however so it's not that big of deal.

**Comment by KyleAMathews** [comment]
Spent some time trying to debug this and… am not sure what happened. It seems like the acf info isn't being pulled anymore.

**Comment by ilyabo** [comment]
Could you elaborate a bit on the possible ways to automate the rebuild from Wordpress?

**Comment by sebastienfi** [comment]
For some reason some plugins were deactivated on our Pantheon.io demo account.
I updated to Wordpress and reinstalled the plugins, the using-wordpress example is working again.

![image](https://user-images.githubusercontent.com/5808108/32050299-941434b6-ba50-11e7-8b03-816bd46df09b.png)

![image](https://user-images.githubusercontent.com/5808108/32050349-c62adc7a-ba50-11e7-8287-15d5282ab4e5.png)

Closing the issue. Reopen if you feel necessary.

Thanks guys!

**Comment by sebastienfi** [comment]
@ilyabo To automate rebuild on WordPress, you will basically have to use hooks. On Netlify, you could easily use Netlify Webhooks to trigger a rebuild. If you need help on that, please create a new issue as this does not concern the primary goal of the current closed issue.

**Comment by ilyabo** [comment]
Thanks Sebastien!

I think it would also make sense to document that these plugins must be installed in Wordpress. Also, not do the ACF related stuff if set to false in the config.

**Comment by owenhoskins** [comment]
@sebastienfi: Would it be possible to provide an export of the ACF fields from dev-gatbsyjswp.pantheonsite.io? I am having trouble figuring out how the query keys relate to the acf field names: `page_builder_post`, `WordPressAcf_post_photo`, `photo`, `WordPressAcf_image_gallery`, `pictures`. Thanks!

**Comment by Kimpo79** [comment]
Not trying to hijack the issue in any way but don't want to create a duplicate either since i think my issue might be related. If not i can move it to it's own thread. 

I just started exploring gatsbyjs and tried to get a site running with WP as data source. The example which is using `dev-gatbsyjswp.pantheonsite.io` works out of the box but changing to a locally hosted WP gives me an error i cannot figure out. 

I'm getting:
<img width="1048" alt="screen shot 2017-11-08 at 4 36 58 pm" src="https://user-images.githubusercontent.com/4668225/32557675-1b64201e-c4a3-11e7-966d-7a36f483e5d9.png">

I can run a query with Graphiql and i do get results. I have limited knowledge when it comes to gatsbyjs and graphql so i might of course be missing something obvious 😁 

![screen shot 2017-11-08 at 4 46 49 pm](https://user-images.githubusercontent.com/4668225/32558199-84aa1cb2-c4a4-11e7-8181-b8eededfbfb5.png)

I did check a bit in gatsby-source-wordpress and will continue digging around a bit to understand more. I did however see a lot of ACF stuff there which to me looks like it does not really belong in a core plugin like that. This is a WP plugin(one of the more popular but certainly not the only one) that enables addition of custom fields to posts and hard coding these kinds of plugin compatibility features to a core gatsby plugin will probably lead to a lot of headaches in the future.

But, regarding my question, if anyone can point out what i am doing wrong i would really appreciate it :)
